plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceArrowToPageScanner.java

```diff
@@ -0,0 +1,213 @@
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.DateType.DATE;
+import static io.trino.spi.type.IntegerType.INTEGER;
+import static io.trino.spi.type.TimeType.TIME_MICROS;
+import static io.trino.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static org.apache.arrow.vector.complex.BaseRepeatedValueVector.OFFSET_WIDTH;
+
+
+public class LanceArrowToPageScanner implements AutoCloseable {
+    private final BufferAllocator allocator;
+    private final List<Type> columnTypes;
+    private final List<String> columnNames;
+    private final Dataset lanceDataset;
+    private final LanceScanner lanceScanner;
+    private final ArrowReader arrowReader;
+    private final VectorSchemaRoot vectorSchemaRoot;
+
+    public LanceArrowToPageScanner(BufferAllocator allocator, String path, List<LanceColumnHandle> columns) {
+        this.allocator = requireNonNull(allocator, "allocator is null");
+        this.columnTypes = requireNonNull(columns, "columns is null").stream().map(LanceColumnHandle::trinoType)
+                .collect(toImmutableList());
+        this.columnNames = columns.stream().map(LanceColumnHandle::name).collect(toImmutableList());
+        try {
+            this.lanceDataset = Dataset.open(path, allocator);
+            // TODO: add additional configs such as batch size, column, filter, etc.
+            this.lanceScanner = lanceDataset.newScan();
+            this.arrowReader = lanceScanner.scanBatches();
+            this.vectorSchemaRoot = arrowReader.getVectorSchemaRoot();
+        } catch (IOException e) {
+            throw new RuntimeException("Unalbe to initialize lance DB connection", e);
+        }
+    }
+
+    public boolean read() {
+        try {
+            return arrowReader.loadNextBatch();
+        } catch (IOException e) {
+            throw new RuntimeException("Error loading next batch!", e);
+        }
+    }
+
+    public void convert(PageBuilder pageBuilder) {
+        pageBuilder.declarePositions(vectorSchemaRoot.getRowCount());
+        List<FieldVector> fieldVectors = vectorSchemaRoot.getFieldVectors();
+
+        for (int column = 0; column < columnTypes.size(); column++) {
+            convertType(pageBuilder.getBlockBuilder(column), columnTypes.get(column), fieldVectors.get(column), 0,
+                    fieldVectors.get(column).getValueCount());
+        }
+        vectorSchemaRoot.clear();
+    }
+
+    private void convertType(BlockBuilder output, Type type, FieldVector vector, int offset, int length) {
+        Class<?> javaType = type.getJavaType();
+        try {
+            if (javaType == boolean.class) {
+                writeVectorValues(output, vector,
+                        index -> type.writeBoolean(output, ((BitVector) vector).get(index) == 1), offset, length);
+            } else if (javaType == long.class) {
```

Comment:
How does Trino present "Vector"?

In lance, a vector is `arrow.FixedSizeList<float16/float32/float64>`

Reply:
Array i think